Description: Preserve Debian packaging integration against the snapshot sdist.
Forwarded: not-needed
Last-Update: 2026-09-29

--- a/setup.cfg
+++ b/setup.cfg
@@ -1,6 +1,12 @@
 [metadata]
 name = heat-tempest-plugin
 
+[entry_points]
+tempest.test_plugins =
+	heat = heat_tempest_plugin.plugin:HeatTempestPlugin
+oslo.config.opts =
+	heat-tempest-plugin = heat_tempest_plugin.config:list_opts
+
 [egg_info]
 tag_build = 
 tag_date = 0
